Earthfile

```diff
@@ -104,9 +104,17 @@ earthly-script-no-stdout:
     FROM earthly/dind:alpine
     RUN apk add --no-cache --update bash
     COPY earthly .earthly_version_flag_overrides .
+
+    # TODO this script performs an explicit "docker pull earthlybinaries:prerelease" which can cause rate-limiting
+    # to work-around this, we will copy an earthly binary in, and update the last-check state files
+    COPY +for-linux/earthly /root/.earthly/earthly-prerelease
+    RUN echo "$(date +%s)" >/tmp/last-earthly-prerelease-check
+    RUN echo "$(md5sum .earthly_version_flag_overrides | awk '{print $1}')" > /tmp/last-earthly-flag-hash
```

Comment:
Feels a bit hacky - could we add a flag in `./earthly` that disables auto-updates instead?

Reply:
That's a good call, I added in a `EARTHLY_DISABLE_AUTO_UPDATE=true` to disable it. It seems more robust than having to pop flags (and not have to worry about it showing up in the middle of the regular earthly flags).